website/src/content/api-reference/commercetools-frontend-application-shell-connectors.mdx

```diff
@@ -0,0 +1,317 @@
+
+This is useful to inject values specific to your application. For example:
+
+```json title="custom-application-config.json" highlightLines="5-7"
+{
+  "name": "My Custom Application",
+  "entryPointUriPath": "my-custom-application",
+  "cloudIdentifier": "gcp-eu",
+  "additionalEnv": {
+    "trackingSentry": "https://000@sentry.io/000",
+  }
+}
+```
+
+The `context.environment` object will then include the `trackingSentry` property:
+
+```json highlightLines="6"
+{
+  "environment": {
+    "applicationId": "ckw3vt1hv034901uzio4bkzc3:my-custom-application",
+    "applicationName": "My Custom Application",
+    "entryPointUriPath": "my-custom-application",
+    "trackingSentry": "https://000@sentry.io/000",
+  }
+}
+```
+
+## Usage
+
+### useApplicationContext
+
+A React hook that allows to access the `ApplicationContext` and selectively pick the necessary properties.
+
+```jsx highlightLines="4-6"
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+
+const DisplayLocale = () => {
+  const dataLocale = useApplicationContext((context) =>
+    applicationContext.dataLocale
+  );
+  render() {
+    return (
+      <div>
+        <h1>Current data locale: {dataLocale}</h1>
+      </div>
+    );
+  }
+}
+
+export default DisplayLocale;
+```
+
+### withApplicationContext
+
+A higher-order component (HOC) that allows to access the `ApplicationContext` and selectively pick the necessary properties. This is the equivalent of the React hook `useApplicationContext`.
+
+```jsx highlightLines="13-15"
+import { withApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+
+class DisplayLocale extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1>Current data locale: {this.props.dataLocale}</h1>
+      </div>
+    );
+  }
+}
+
+export default withApplicationContext((applicationContext) => ({
+  dataLocale: applicationContext.dataLocale
+}))(DisplayLocale);
+```
+
+# Project image settings
+
+Product images can be uploaded to the commercetools platform or can be referenced from external sources.
+
+By default, images referenced from external sources are not displayed in the Merchant Center. This is primarily due to avoid possible performance issues in case the size of those images is big.
+
+In order to show images referenced from external sources, you need to define a configuration to rewrite the URL of the images to versions of the image that has a smaller size. You can configure that in the Merchant Center **Settings > Project Settings > Miscellaneous**.
+
+This configuration can be fetched from your Custom Application using the following components.
+
+## Usage
+
+### ProjectExtensionProviderForImageRegex
+
+This is the React context provider that loads the image regex configuration and expose it in the React context so that children can access the configuration.
```

Comment:
```suggestion
This is the React context provider that loads the image regex configuration and exposes via a React context so that children can access the configuration.
```